crud/select/plan.lua

```diff
@@ -15,19 +15,23 @@ local function index_is_allowed(index)
 end
 
 local function get_index_for_condition(space_indexes, space_format, condition)
-    for i= 0, #space_indexes do
+    for i= 0, table.maxn(space_indexes) do
```

Comment:
Hi! Thanks for your patch!

Could you move length calculation into separate variable that could be used here and below?
I'll place some benchmarks below. It's too expensive to call `table.maxn` twice.

Also could you add a space between "i" and "=" `i= 0` -> `i = 0` since you change this place?

```lua
local clock = require('clock')

local t = {}
for i = 1, 1e5 do
    t[i] = i
end

local s = clock.time()
local _
for _ = 1, 1e5 do
    _ = table.maxn(t)
end
print('maxn', clock.time() - s)

s = clock.time()

local _
for _ = 1, 1e5 do
    _ = #t
end
print('#', clock.time() - s)

-- tarantool test.lua 
-- maxn	6.1955099105835
-- #	0.012683868408203
```

Reply:
However, my benchmark could be not relevant since amount of indexes is restricted with 65k:

For arrays with length ~1k table.maxn even faster than `#`. But anyway if we could call it only once we should do it.